src/main/java/com/microsoft/azure/functions/worker/handler/FunctionEnvironmentReloadRequestHandler.java

```diff
@@ -0,0 +1,73 @@
+package com.microsoft.azure.functions.worker.handler;
+
+import java.lang.reflect.Field;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.microsoft.azure.functions.rpc.messages.*;
+import com.microsoft.azure.functions.rpc.messages.FunctionEnvironmentReloadResponse.Builder;
+import com.microsoft.azure.functions.worker.broker.JavaFunctionBroker;
+import com.microsoft.azure.functions.worker.description.FunctionMethodDescriptor;
+
+public class FunctionEnvironmentReloadRequestHandler
+    extends MessageHandler<FunctionEnvironmentReloadRequest, FunctionEnvironmentReloadResponse.Builder> {
+  public FunctionEnvironmentReloadRequestHandler(JavaFunctionBroker broker) {
+    super(StreamingMessage::getFunctionEnvironmentReloadRequest,
+        FunctionEnvironmentReloadResponse::newBuilder,
+        FunctionEnvironmentReloadResponse.Builder::setResult,
+        StreamingMessage.Builder::setFunctionEnvironmentReloadResponse);
+
+    this.broker = broker;
+  }
+
+  @Override
+  String execute(FunctionEnvironmentReloadRequest request, Builder response) throws Exception {
+    EnvironmentVariables = request.getEnvironmentVariablesMap();
+    if (EnvironmentVariables == null || EnvironmentVariables.isEmpty()) {    
+      return String.format("Ignoring FunctionEnvironmentReloadRequest as newSettings map is either empty or null");
+    }
+    
+    setEnv(EnvironmentVariables);    
+    return String.format("FunctionEnvironmentReloadRequest completed");
+  }
+
+  FunctionMethodDescriptor createFunctionDescriptor(String functionId,
+      RpcFunctionMetadata metadata) {
+    return new FunctionMethodDescriptor(functionId, metadata.getName(), metadata.getEntryPoint(),
+        metadata.getScriptFile());
+  }
+  
+  public static void setEnv(Map<String, String> newSettings) throws Exception {
```

Comment:
@fabiocav - Java does not have a built in helper to set the environment variables. Following code snippet is taken from [SO post](https://stackoverflow.com/questions/318239/how-do-i-set-environment-variables-from-java)

Reply:
Wow... this looks pretty fragile. Should we have a note here?